cmd/noms/splore/noms_splore.go

```diff
@@ -0,0 +1,307 @@
+		if children == nil {
+			children = []nodeChild{}
+		}
+	case types.List:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			v.IterAll(func(vi types.Value, i uint64) {
+				children[i] = nodeChild{
+					Value: info(vi, fmt.Sprintf("%s[%d]", parentPath, i)),
+				}
+			})
+		}
+	case types.Map:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			i := 0
+			v.IterAll(func(k, v types.Value) {
+				children[i] = nodeChild{
+					Key:   info(k, atPath(i, "@key")),
+					Value: info(v, atPath(i, "")),
+				}
+				i++
+			})
+		}
+	case types.Set:
+		children = getMetaChildren(v)
+		if children == nil {
+			children = make([]nodeChild, v.Len())
+			i := 0
+			v.IterAll(func(v types.Value) {
+				children[i] = nodeChild{
+					Value: info(v, atPath(i, "")),
+				}
+				i++
+			})
+		}
+	case types.Ref:
+		children = []nodeChild{{
+			Value: info(v, parentPath+"@target"),
+		}}
+	case types.Struct:
+		children = make([]nodeChild, v.Len())
+		i := 0
+		v.IterFields(func(name string, v types.Value) {
+			children[i] = nodeChild{
+				Label: name,
+				Value: info(v, fmt.Sprintf("%s.%s", parentPath, name)),
+			}
+			i++
+		})
+	default:
+		panic(fmt.Errorf("unsupported value type %T", v))
+	}
+	return
+}
+
+func nodeName(v types.Value) string {
+	typeName := func(iface interface{}) string {
+		return reflect.TypeOf(iface).Name()
+	}
+
+	switch v := v.(type) {
+	case types.Bool, types.Number, types.String:
+		return fmt.Sprintf("%#v", v)
+	case types.Blob:
+		return fmt.Sprintf("%s(%s)", typeName(v), humanize.Bytes(v.Len()))
+	case types.List, types.Map, types.Set:
+		return fmt.Sprintf("%s(%d)", typeName(v), v.(types.Collection).Len())
+	case types.Ref:
+		kind := v.TargetType().Desc.Kind()
+		return fmt.Sprintf("%s#%s", kind.String(), v.TargetHash().String())
+	case types.Struct:
+		if v.Name() == "" {
+			return "Struct{…}"
+		}
+		return v.Name()
+	}
+	panic("unreachable")
+}
+
+// getMetaChildren returns the nodeChild children, as refs, of v if it's backed
+// by a meta sequence, or nil if not.
+//
+// This isn't exposed directly on the API but for now just guess it:
+// - If there are no chunks, it must be a leaf.
+// - If there are MORE chunks than the length of the blob/collection then it
+//   can only be a leaf with multiple ref values per entry.
+// - If there are EQUAL then it could be either, but heuristically assume
+//   that it's a leaf with a ref value per entry. It's highly unlikely that a
+//   blob/collection will chunk with single elements.
+// - If there are LESS then it could be either a chunked blob/collection or a
+//   collection of mixed types, but heuristically assume that's it's chunked.
+func getMetaChildren(v types.Value) (children []nodeChild) {
+	var l uint64
+	if col, ok := v.(types.Collection); ok {
+		l = col.Len()
+	} else {
+		l = v.(types.Blob).Len()
+	}
+
+	vKind := types.TypeOf(v).Desc.Kind()
```

Comment:
This is doing a lot of extra work that is not needed. It has to create the `types.Type`. Just do `v.Kind()`.